tests/test_pipeline/test_schedule/test_zerobubble_pp.py

```diff
@@ -0,0 +1,647 @@
+            # chunk 1 bwd
+            ScheduledNode(type="RECV_BACKWARD", chunk=1, stage=3, minibatch=3),
+            ScheduledNode(type="B", chunk=1, stage=3, minibatch=3),
+            ScheduledNode(type="W", chunk=1, stage=3, minibatch=3),
+            ScheduledNode(type="SEND_BACKWARD", chunk=1, stage=3, minibatch=3),
+            # chunk 0 bwd
+            ScheduledNode(type="RECV_BACKWARD", chunk=0, stage=3, minibatch=3),
+            ScheduledNode(type="B", chunk=0, stage=3, minibatch=3),
+            ScheduledNode(type="W", chunk=0, stage=3, minibatch=3),
+            ScheduledNode(type="SEND_BACKWARD", chunk=0, stage=3, minibatch=3),
+        ],
+    ]
+
+    scheduler = ZeroBubbleVPipeScheduler(
+        schedule=zbv_schedule[rank],  # hint: send whole schedule or local schedule only ?
+        stage_manager=stage_manager,
+        num_model_chunks=pp_size,
+        num_microbatch=num_microbatch,
+        overlap_p2p=False,
+    )
+
+    # loss func
+    def criterion(x, *args, **kwargs):
+        return (x * x).mean()
+
+    # init model and input
+    batch_size = 4
+    num_layers = 8
+    in_dim = out_dim = 8
+    print(f"Before init Model: {torch.cuda.memory_allocated()/1024**3 :.3f} GB on device {stage_manager.get_rank()};")
+    model = MlpModel(in_dim=in_dim, out_dim=out_dim, num_layers=num_layers).to(rank)
+    data_iter = [torch.rand(batch_size, in_dim, out_dim, requires_grad=True).to(rank)]
+
+    input_base = [t.clone() for t in data_iter]
+    model_base = deepcopy(model)
+
+    if rank == 0:
+        # layer 0 & 7 to chunk 0 on rank0
+        local_chunk = torch.nn.ModuleList().to(rank)
+        for idx, sub_model in enumerate(model.layers):
+            if idx == 0 or idx == 7:
+                local_chunk.append(sub_model)
+    elif rank == 1:
+        # layer 1 & 6 to chunk 1 on rank1
+        local_chunk = torch.nn.ModuleList().to(rank)
+        for idx, sub_model in enumerate(model.layers):
+            if idx == 1 or idx == 6:
+                local_chunk.append(sub_model)
+    elif rank == 2:
+        # layer 2 & 5 to chunk 2 on rank2
+        local_chunk = torch.nn.ModuleList().to(rank)
+        for idx, sub_model in enumerate(model.layers):
+            if idx == 2 or idx == 5:
+                local_chunk.append(sub_model)
+    else:
+        # layer 3 & 4 to chunk 3 on rank3
+        local_chunk = torch.nn.Sequential().to(rank)
+        for idx, sub_model in enumerate(model.layers):
+            if idx == 3 or idx == 4:
+                local_chunk.append(sub_model)
+    print(
+        f"After init Model & input: {torch.cuda.memory_allocated()/1024**3 :.3f} GB on device {stage_manager.get_rank()};"
+    )
+
+    torch.cuda.synchronize()
+    scheduler.run_forward_backward(
+        model_chunk=local_chunk,
+        data_iter=iter(data_iter),
+        criterion=criterion,
+        optimizer=None,
+        return_loss=None,
+        return_outputs=None,
```

Comment:
Should also check that outputs are all-close?

Reply:
> Should also check that outputs are all-close?

Done in latest push https://github.com/hpcaitech/ColossalAI/pull/6034/commits/6c2a120bed8658015f0f4e4ee95cbbe314b6ce5e;